test/cpp/end2end/grpclb_end2end_test.cc

```diff
@@ -604,51 +626,23 @@ TEST_F(SingleBalancerTest, RepeatedServerlist) {
   ScheduleResponseForBalancer(
       0, BalancerServiceImpl::BuildResponseForBackends(GetBackendPorts(), {}),
       kServerlistDelayMs);
-
   // Send num_backends/2 requests.
-  auto statuses_and_responses = SendRpc(kMessage_, num_backends_ / 2);
-  // only the first half of the backends will receive them.
+  CheckRpcSendOk(num_backends_ / 2);
+  // only half of the backends will receive them.
+  size_t num_requests = 0;
   for (size_t i = 0; i < backends_.size(); ++i) {
-    if (i < backends_.size() / 2)
-      EXPECT_EQ(1U, backend_servers_[i].service_->request_count())
-          << "for backend #" << i;
-    else
-      EXPECT_EQ(0U, backend_servers_[i].service_->request_count())
-          << "for backend #" << i;
-  }
-  EXPECT_EQ(statuses_and_responses.size(), num_backends_ / 2);
-  for (const auto& status_and_response : statuses_and_responses) {
-    const Status& status = status_and_response.first;
-    const EchoResponse& response = status_and_response.second;
-    EXPECT_TRUE(status.ok()) << "code=" << status.error_code()
-                             << " message=" << status.error_message();
-    EXPECT_EQ(response.message(), kMessage_);
+    num_requests += backend_servers_[i].service_->request_count();
   }
-
   // Wait for the (duplicated) serverlist update.
-  gpr_sleep_until(gpr_time_add(
-      gpr_now(GPR_CLOCK_REALTIME),
-      gpr_time_from_millis(kServerlistDelayMs * 1.1, GPR_TIMESPAN)));
-
+  gpr_sleep_until(grpc_timeout_milliseconds_to_deadline(kServerlistDelayMs));
   // Verify the LB has sent two responses.
   EXPECT_EQ(2U, balancer_servers_[0].service_->response_count());
-
   // Some more calls to complete the total number of backends.
-  statuses_and_responses = SendRpc(
-      kMessage_,
-      num_backends_ / 2 + (num_backends_ & 0x1) /* extra one if num_bes odd */);
+  CheckRpcSendOk(num_backends_ / 2 + 1);
   // Because a duplicated serverlist should have no effect, all backends must
-  // have been hit once now.
+  // have been hit at least once now.
```

Comment:
How do we know that all backends have been hit?  We haven't verified that they've all become ready.

Reply:
Where do we check that all backends have a count of at least 1?  I don't see that anywhere.

Note that in the loop above, we're setting `num_requests` to the total number of requests fielded by all backends, but we're not even bothering to check the resulting value.  And even if we did, that would only tell us that the right total number of requests was seen across all backends; it would not tell us anything about the distribution of those requests across the backends.